### ISSUE TITLE
logs: DEV channel events are unexpectedly appearing in custom file-groups

### ISSUE THREAD

**Issue by thtruo** [cockroachdb/cockroach#59004]
**Observation**:

I'm seeing DEV channel events appear in custom file-groups that do not explicitly include DEV. 

**Repro**: 
I intend to only capture SQL_SCHEMA channel events in one of my sinks
`cockroach start-single-node --certs-dir=certs --log='sinks: {file-groups: {ghdev: {channels: SQL_SCHEMA, format: json}}}'
`
However, inspecting the log shows DEV channel events too. That was unexpected.

Example output from the tail of that log file 
[cockroach-ghdev.TommyCocBookPro.tommytruongchau.2021-01-14T16_58_30Z.023129.log](https://github.com/cockroachdb/cockroach/files/5815849/cockroach-ghdev.TommyCocBookPro.tommytruongchau.2021-01-14T16_58_30Z.023129.log)

cc @piyush-singh

**Comment by piyush-singh** [comment]
I ran into this too as I noted [here](https://github.com/cockroachdb/cockroach/issues/58970#issuecomment-760287396). The DEV channel events were logged even with a severity filter that should have excluded them.

**Comment by thtruo** [comment]
Gotcha. I checked a few more examples and am seeing DEV events sprinkled throughout various file-groups. 

For example in this config (below), I'm unexpectedly seeing DEV events in `cockroach-health.log` and `cockroach-security.log`. Am I misunderstanding something or does this strike you as a bug? cc @knz 

```
file-defaults:
  dir: cockroach-data/logs 
  filter: INFO             
  redact: false          
  redactable: false   
  max-file-size: 10MB  
  max-group-size: 100MB  
  format: json             

sinks:
 file-groups:              
   security:               # cockroach-security.log
     channels: "SESSIONS, USER_ADMIN, PRIVILEGES, SENSITIVE_ACCESS"
     format: json
   health:                 # cockroach-health.log
     channels: "HEALTH, OPS"
     format: json
   devonly:                # cockroach-devonly.log
     channels: "DEV"
     format: json
~                        
```

Here's an accompany [visualization](http://www.plantuml.com/plantuml/uml/T9FFRvim5CVl_YiiJjiXHVdH7dQOYXAlGKgJB4wZ7IOXonZ2Plmo_wWKJVdV9ni89MjFlFTz7uomFtuqbcdhQecaA2osW7LL7YtEAosuhK0X3lK9b50M6t2Q2uF_ea-VyO8S_6EpfVwn9Cbgl_GLtMzsoHC99Q4qtQm3GB-lCZfVakV4To-KxB9ayPokVRVTfOTqHPv8Gv8rJVVfWMJ9V4xezMtoWyoxUajstxewNU_9Rfsq8PVWScoiPcVC38v2zLC9hK6R25tG7oUSmCRcGcl4cSxnASl67LkTeciu2M41kcRsooy3gXzDt-Dd7_Ftx_4F__DCswfWtEAYac6ZSG4o5nf7yOiHsiGMwlfijGDLyYDpyGAuguMo9Zv9TYurE9N7hvFntJYJK6QEsPW3_ws1yUCePvR54aeJ-PN3Dt3H_FeRIBo2akSx2MLqXIOtq54mQOz3Pdh36C6ThknmfMRVGtb3Xq7BNj05BSW1ZqPVmq6WLgetmLMdOUOb6IRUYc4o4AOV1GryLunHEFrsCaQRBUsQ6MfCx_jWGtWjB37h9bFKwTyb3wZNlizw-VlizWhqa-1I-DeK1ONQoJrgbFTjlyVeKQZSt-r_0G00__y0) of the setup:
![image](https://user-images.githubusercontent.com/1474303/104662452-d98f5880-5698-11eb-9d6a-6b7c62f6c14c.png)

**Comment by knz** [comment]
what y'all are observing are the file headers, which are printed out at the start of every log file with channel DEV no matter the filtering rules and channel selection.

We need to keep these headers, but it's silly to have them attached to a channel given that their are attached to files, not channels.

**Comment by knz** [comment]
this will be addressed/clarified in #59064
